sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2082,6 +2083,65 @@ case class UnBase64(child: Expression)
   }
 }
 
+object Decode {
+  def createExpr(params: Seq[Expression]): Expression = {
+    params.length match {
+      case 0 | 1 =>
+        throw new AnalysisException("Invalid number of arguments for function decode. " +
+          s"Expected: 2; Found: ${params.length}")
+      case 2 => StringDecode(params.head, params.last)
```

Comment:
hm, IMHO it would be good to rename the existing `decode` func or assign a new name into the new `decode` func because the generated doc for `decode` will be complicated.

Reply:
If rename the `existing` decode func will break the behavior.

Reply:
Yea, so I think we need add a legacy config, update the migration guide, ... if we rename it. Anyway, the latter approach (assigning a new name into the new `decode` func) looks better if possible.

Reply:
This PR could avoid migration and support the new `decode` as well. cc @cloud-fan

Reply:
We can probably rename the function in ansi mode.